compass/landice/mesh.py

```diff
@@ -753,3 +764,344 @@ def make_region_masks(self, mesh_filename, mask_filename, cores, tags):
+    for field in ['thk', 'bheatflx', 'vx', 'vy',
+                  'ex', 'ey', 'thkerr', 'dhdt']:
+        tic = time.perf_counter()
+        logger.info('Beginning building interpolator for {}'.format(field))
+        if field in ['thk', 'thkerr']:
+            mask = cellsWithIce.ravel()
+        elif field == 'bheatflx':
+            mask = np.logical_and(
+                data.variables[field][:].ravel() < 1.0e9,
+                data.variables[field][:].ravel() != 0.0)
+        elif field in ['vx', 'vy', 'ex', 'ey', 'dhdt']:
+            mask = np.logical_and(
+                data.variables[field][:].ravel() < 1.0e9,
+                cellsWithIce.ravel() > 0)
+        else:
+            mask = cellsWithIce
+        interp = NearestNDInterpolator(
+            list(zip(xx[mask], yy[mask])),
+            data.variables[field][:].ravel()[mask])
+        toc = time.perf_counter()
+        logger.info('Finished building interpolator in {} seconds'.format(
+            toc - tic))
+
+        tic = time.perf_counter()
+        logger.info('Beginning interpolation for {}'.format(field))
+        data.variables[field][0, :] = interp(xGrid, yGrid)
+        toc = time.perf_counter()
+        logger.info('Interpolation completed in {} seconds'.format(
+            toc - tic))
+
+    bigToc = time.perf_counter()
+    logger.info('All interpolations completed in {} seconds'.format(
+        bigToc - bigTic))
+
+    # Now perform some additional clean up adjustments to the dataset
+    data.createVariable('dHdtErr', 'f', ('time', 'y1', 'x1'))
+    data.variables['dHdtErr'][:] = dHdtErr
+
+    data.createVariable('vErr', 'f', ('time', 'y1', 'x1'))
+    data.variables['vErr'][:] = np.sqrt(data.variables['ex'][:]**2 +
+                                        data.variables['ey'][:]**2)
+
+    data.variables['bheatflx'][:] *= -1.e-3  # correct units and sign
+    data.variables['bheatflx'].units = 'W m-2'
+
+    data.variables['subm'][:] *= -1.0  # correct basal melting sign
+    data.variables['subm_ss'][:] *= -1.0
+
+    data.renameVariable('dhdt', 'dHdt')
+    data.renameVariable('thkerr', 'topgerr')
+
+    data.createVariable('x', 'f', ('x1'))
+    data.createVariable('y', 'f', ('y1'))
+    data.variables['x'][:] = x1
+    data.variables['y'][:] = y1
+
+    data.close()
+
+    return preprocessed_gridded_dataset
+
+
+def interp_ais_bedmachine(self, data_path, mali_scrip, nProcs, dest_file):
+    """
+    Interpolates BedMachine thickness and bedTopography dataset
+    to a MALI mesh
+
+    Parameters
+    ----------
+    data_path : str
+        path to AIS datasets, including BedMachine
+
+    mali_scrip : str
+        name of scrip file corresponding to destination MALI mesh
+
+    nProcs : int
+        number of processors to use for generating remapping weights
+
+    dest_file: str
+        MALI input file to which data should be remapped
+    """
+
+    logger = self.logger
+
+    logger.info('creating scrip file for BedMachine dataset')
```

Comment:
This step is really slow, so we should check for an existing scrip file before proceeding. It should also write to the compass working directory instead of to the data directory if it does create a new scrip file. I'd rather have the data directory untouched.

Reply:
Can you check if your compass env contains the changes to MPAS-Tools here? https://github.com/MPAS-Dev/MPAS-Tools/pull/544

This step got like 100x faster after that change.  I'm wondering if you'd be comfortable skipping the check for an existing file if this step only takes a minute.

Reply:
Actually I think I know the answer - the conda env version for this branch would not have the MPAS-Tools update in it, because those were made at the same time as this PR.  But I believe that MPAS-Tools PR is part of the current compass env.  So try redoing your test after merging this branch locally into main (or rebasing) and creating an updated compass env.

Reply:
I've modified to write the scrip file to the workdir.